### PR TITLE
Test MetricSerializer, EventSerializer and ServiceCheckSerializer

### PR DESCRIPTION
**Pull request by ogaca-dd**
This PR splits the class `StatsdUnitTests` to test `MetricSerializer`, `EventSerializer` and `ServiceCheckSerializer`.

For reviewers:
- There are few new additions in this PR and most of the changes are code moved / refactored.
**Reviewing commit by commit is recommended to be time efficient.**
- `EventSerializer`, `ServiceCheckSerializer` and `MetricSerializer` have some old code from  `Statsd` but they will be completely refactored in the next PR.

Can be merged when https://github.com/DataDog/dogstatsd-csharp-client/pull/123 is merged.

### CODE REVIEW

File: src/StatsdClient/Serializer/EventSerializer.cs [hunk 1]
```diff
@@ -47,7 +53,7 @@ public static string GetCommand(string title, string text, string alertType, str
                 result += string.Format(CultureInfo.InvariantCulture, "|t:{0}", alertType);
             }
 
-            result += MetricSerializer.ConcatTags(constantTags, tags);
+            result += MetricSerializer.ConcatTags(_constantTags, tags);
 
             if (result.Length > MaxSize)
```
Comment (ogaca-dd) [inline src/StatsdClient/Serializer/EventSerializer.cs:58]: https://github.com/DataDog/dogstatsd-csharp-client/pull/124/commits/88643cf170dad14e90a706f117980fe0349ccb3a#diff-2b2429c2b7220221984fa0ab16b8924eL66: 
In the previous version, when truncating the metric, `GetCommand` overload was called without `constantTags` which was an issue.  As `_constantTags` is a member of EventSerializer, `constantTags` is always used in the new version.

File: src/StatsdClient/Serializer/ServiceCheckSerializer.cs [hunk 2]
```diff
@@ -29,7 +35,7 @@ public static string GetCommand(string name, int status, int? timestamp, string
                 result += string.Format(CultureInfo.InvariantCulture, "|h:{0}", hostname);
             }
 
-            result += MetricSerializer.ConcatTags(constantTags, tags);
+            result += MetricSerializer.ConcatTags(_constantTags, tags);
```
Comment (ogaca-dd) [inline src/StatsdClient/Serializer/ServiceCheckSerializer.cs:38]: Same as https://github.com/DataDog/dogstatsd-csharp-client/pull/124/commits/88643cf170dad14e90a706f117980fe0349ccb3a#r432316492.

File: src/StatsdClient/Serializer/ServiceCheckSerializer.cs [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                if (!truncateIfTooLong)
+                {
+                    throw new Exception(string.Format("ServiceCheck {0} payload is too big (more than 8kB)", name));
+                }
+
+                var overage = result.Length - MaxSize;
+
+                if (processedMessage == null || overage > processedMessage.Length)
+                {
+                    throw new ArgumentException(string.Format("ServiceCheck name is too long to truncate, payload is too big (more than 8Kb) for {0}", name), "name");
```
Comment (truthbk) [inline src/StatsdClient/Serializer/ServiceCheckSerializer.cs:57]: 8Kb is already going to be fragmented on scenarios without jumbo frames.... This works well for `localhost` but may be a problem for remote submission. I know this was already the case, but it made me think.

Comment (ogaca-dd) [reply src/StatsdClient/Serializer/ServiceCheckSerializer.cs:57]: I have started a conversation at https://github.com/DataDog/dogstatsd-csharp-client/pull/123#discussion_r455489491.

File: src/StatsdClient/StatsdBuilder.cs [hunk 4]
```diff
@@ -1,5 +1,6 @@
 using System;
 using System.Collections.Generic;
+using System.Linq;
```
[the remark this answers is not in the document]
Comment (ogaca-dd) [reply src/StatsdClient/StatsdBuilder.cs:3]: It is used by the extension method `ToArray` https://github.com/DataDog/dogstatsd-csharp-client/blob/2d27f29952ceacc1876ace5c14fc0a2b40607c78/src/StatsdClient/StatsdBuilder.cs#L63

`using System.Linq` is a namespace that contains extension methods for `IEnumerable` interface. In C# you can add a public method to an existing class with method extensions if the method extension only uses public methods of the class.

**Pull request merged**